app/server/credential/src/main/java/io/syndesis/server/credential/SocialProperties.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.syndesis.server.credential;
+
+/**
+ * Basic configuration properties for spring social.
+ */
+public class SocialProperties {
```

Comment:
I understand you're cloning this [java bean](https://docs.spring.io/spring-boot/docs/2.0.0.M3/api/org/springframework/boot/autoconfigure/social/SocialProperties.html) to avoid bringing in another spring library, isn't it?

Reply:
Yes Spring Boot 2.x has removed that bean and we kinda use it heavily so this was the best way for me to align our code with newer Spring versions. Maybe requires some further refactoring to make our code also not depend on that particular class in future.

Reply:
It's a long standing problem that we depend on a forked version of `spring-social` (e.g. #5102), we could (at a later point in time) refactor this code so we don't depend on it at all.